src/charm.py

```diff
@@ -366,31 +374,27 @@ def _make_env_config(self, domain="svc.cluster.local") -> dict:
         """
         config = self.model.config
         relation = self.model.get_relation("ingress-proxy")
-        if relation is not None and relation.data[relation.app]:
-            site = relation.data[relation.app]["service-hostname"]
-            svc_name = relation.data[relation.app]["service-name"]
-            svc_port = relation.data[relation.app]["service-port"]
-            backend_site_name = relation.data[relation.app]["service-hostname"]
-            clients = []
-            for peer in relation.units:
-                unit_name = peer.name.replace("/", "-")
-                service_url = f"{unit_name}.{svc_name}-endpoints.{self.model.name}.{domain}"
-                clients.append(f"http://{service_url}:{svc_port}")
-            # XXX: Will need to deal with multiple units at some point
-            backend = clients[0]
-        elif relation:
-            ingress_config = self._make_ingress_config()
-            site = "localhost"
-            svc_name = ingress_config["service-name"]
-            svc_port = ingress_config["service-port"]
-            backend = f"http://{site}:{svc_port}"
-            backend_site_name = urlparse(backend).hostname
-        else:
-            backend = config["backend"]
-            backend_site_name = config.get("backend_site_name")
-            if not backend_site_name:
-                backend_site_name = urlparse(backend).hostname
-            site = config["site"]
+        try:
+            if relation is not None and relation.data[relation.app]:
+                site = relation.data[relation.app]["service-hostname"]
```

Comment:
Can't you use `get` instead and remove the try-catch block?

Reply:
I'd prefer to stick with the  try-catch block, since there are too many things to check with the if/get clauses (we'd have to check for the existance of every field like service-hostname, backend, service-name, service-port, etc) and I think it would be computationally more expensive to do all of those.

Reply:
`site = relation.data[relation.app].get("service-hostname")` would me more compact, more testable and more readable. As a general rule, it's better to use exceptions for things that are not expected to happen. Is this the case? Are all those values supposed to be always there in the data bag or are they optional? If it is not abnormal it would be better to avoid catching an exception, if it is, why don't left the charm fall into error state?